pkg/platform/abstract/platform.go

```diff
@@ -469,3 +488,54 @@ func (ap *Platform) prettifyProcessorLogLine(log []byte) (string, bool, error) {
 
 	return res, false, nil
 }
+
+// Function must have project name - if it was not given - set to default project
+func (ap *Platform) enrichProjectName(createFunctionOptions *platform.CreateFunctionOptions) error {
+
+	// if labels is nil assign an empty map to it
+	if createFunctionOptions.FunctionConfig.Meta.Labels == nil {
+		createFunctionOptions.FunctionConfig.Meta.Labels = map[string]string{}
+	}
+
+	// if no project name was given, set it to the default project
+	if createFunctionOptions.FunctionConfig.Meta.Labels["nuclio.io/project-name"] == "" {
+		createFunctionOptions.FunctionConfig.Meta.Labels["nuclio.io/project-name"] = "default"
+		ap.Logger.Debug("No project name specified. Setting to default")
+	}
+
+	return nil
+}
+
+// If a user specify the image name to be built - add "projectName-functionName-" prefix to it
+func (ap *Platform) enrichImageName(createFunctionOptions *platform.CreateFunctionOptions) error {
+	if ap.ImageNamePrefixTemplate == "" {
+		return nil
+	}
+	functionName := createFunctionOptions.FunctionConfig.Meta.Name
+	projectName := createFunctionOptions.FunctionConfig.Meta.Labels["nuclio.io/project-name"]
+
+	functionBuildRequired, err := ap.functionBuildRequired(createFunctionOptions)
+	if err != nil {
+		return errors.Wrap(err, "Failed determining whether function should build")
+	}
+
+	// if build is not required or custom image name was not asked enrichment is irrelevant
+	if !functionBuildRequired || createFunctionOptions.FunctionConfig.Spec.Build.Image == "" {
```

Comment:
Add comment that leaving this empty will cause further enrichment down the line, deeper (in `build/builder.go`) - we need to revisit the need for this logic being stretched on so many places :|